core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -406,49 +409,37 @@ private[netty] class NettyRpcHandler(
   private type RemoteEnvAddress = RpcAddress
 
   // Store all client addresses and their NettyRpcEnv addresses.
+  // TODO: Is this even necessary?
   @GuardedBy("this")
   private val remoteAddresses = new mutable.HashMap[ClientAddress, RemoteEnvAddress]()
```

Comment:
Yes. It's necessary. `RemoteEnvAddress` is the port that a `RpcEnv` listens to, but `ClientAddress` is a random port that the client opens. For these network events, `RpcEndpoint` needs to use `RemoteEnvAddress`.